### PR TITLE
restructured arg parsing and processing to utils

### PR DESCRIPTION
**Pull request by gerazov**
I've moved the command line parsing and processing from the `if __name__ == '__main__':` block into two separate functions in a new file `TTS/utils/arguments.py`. This avoids code douplication across the training scripts adhering to the DRY principle :slightly_smiling_face: 

It also solves the `FIXME` that was before the `main` function.

Currently updated:
- `train_glow_tts.py`
- `train_speedy_speech.py`
- `train_tacotron.py`
- `train_vocoder_gan.py`
- `train_vocoder_wavegrad.py`
- `train_vocoder_wavernn.py`

These still work as before, they are a bit different - but I can update them if necessary:
- `train_encoder.py`
- `tune_wavegrad.py`

### CODE REVIEW

File: TTS/bin/train_glow_tts.py [hunk 1]
```diff
@@ -468,7 +465,6 @@ def evaluate(data_loader, model, criterion, ap, global_step, epoch):
     return keep_avg.avg_values
 
 
-# FIXME: move args definition/parsing inside of main?
```
Comment (gerazov) [inline TTS/bin/train_glow_tts.py:471]: This has to happen outside of main so that the experiment folder can be removed if it fails.

**Comment by gerazov** [comment]
Two tests fail with timeouts :thinking: 

Test 3.8 (360):
```python
 > Testing demo server sentence splitting
... > Initialised Angular Prototypical loss
 > Initialised Angular Prototypical loss
 > Initialised Angular Prototypical loss
. > Initialised Generalized End-to-End loss
 > Initialised Generalized End-to-End loss
 > Initialised Generalized End-to-End loss
... > Num parameters for GlowTTS model:13724545

Too long with no output (exceeded 10m0s): context deadline exceeded
```

Test 3.6 (361)
```python
 > Testing demo server sentence splitting
... > Initialised Angular Prototypical loss
 > Initialised Angular Prototypical loss
 > Initialised Angular Prototypical loss
. > Initialised Generalized End-to-End loss
 > Initialised Generalized End-to-End loss
 > Initialised Generalized End-to-End loss
... > Num parameters for GlowTTS model:13724545

Too long with no output (exceeded 10m0s): context deadline exceeded
```

I've tested all of them locally and Glow TTS training works in my 3.6 based environment.

**Comment by erogol** [comment]
Wow!! Thx for the PR. I think you saved a ton of wasted lines.

There is something wrong with the test environment and sometimes there is a test hanging for too long.
Let me run the tests again.

File: TTS/utils/arguments.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                re.search(r"checkpoint_([0-9]+)", filename).groups()[0])
+            if checkpoint_num > last_checkpoint_num:
+                last_checkpoint_num = checkpoint_num
+                last_checkpoint = filename
+        except AttributeError:  # if there's no match in the filename
+            pass
+    if last_checkpoint is None:
+        raise ValueError(f"No checkpoints in {path}!")
+    else:
+        return last_checkpoint
```
Comment (gerazov) [inline TTS/utils/arguments.py:108]: I've added this for step based model continuing. Currently it doesn't work so I've commented it out in `process_args()` as the best model is not taken into account. The currently used creation time is a bit volatile, so I've started a discussion on this topic in  #643 :wink:

**Comment by gerazov** [comment]
> Wow!! Thx for the PR. I think you saved a ton of wasted lines.
> 
> There is something wrong with the test environment and sometimes there is a test hanging for too long.
> Let me run the tests again.

Thanks :blush: Glad to contribute :+1:

File: TTS/utils/arguments.py (hunk 2 above)
Comment (erogol) [reply TTS/utils/arguments.py:108]: how about if we separate it from this PR? We can discuss that as I merge the rest.

Comment (gerazov) [reply TTS/utils/arguments.py:108]: Yes yes absolutely :+1:

**Pull request merged**